test/IExecWhitelist.test.ts

```diff
@@ -0,0 +1,98 @@
+    IExecWhitelistFactory = await ethers.getContractFactory('IExecWhitelist');
+    [owner, addr1, addr2] = await ethers.getSigners();
+    whitelist = (await IExecWhitelistFactory.deploy()) as IExecWhitelist;
+  });
+
+  describe('Whitelisting', () => {
+    it('should allow the owner to add a dapp address to the whitelist', async () => {
+      await whitelist
+        .connect(owner)
+        .addRessourceToWhitelist(await addr1.getAddress());
+      const paddedAddress =
+        '0x' + '00'.repeat(12) + (await addr1.getAddress()).slice(2);
+      expect(
+        await whitelist.keyHasPurpose(ethers.hexlify(paddedAddress), 4)
+      ).to.equal(true);
+    });
+
+    it('should prevent non-owners from adding a dapp address to the whitelist', async () => {
+      await expect(
+        whitelist
+          .connect(addr1)
+          .addRessourceToWhitelist(await addr2.getAddress())
+      ).to.be.revertedWith('Ownable: caller is not the owner');
+    });
+
+    it('should allow the owner to remove a dapp address from the whitelist', async () => {
+      await whitelist
+        .connect(owner)
+        .addRessourceToWhitelist(await addr1.getAddress());
+      await whitelist
+        .connect(owner)
+        .remRessourceToWhitelist(await addr1.getAddress());
+
+      const paddedAddress =
+        '0x' + '00'.repeat(12) + (await addr1.getAddress()).slice(2);
+      expect(
+        await whitelist.keyHasPurpose(ethers.hexlify(paddedAddress), 4)
+      ).to.equal(false);
+    });
+
+    it('should prevent non-owners from removing a dapp address from the whitelist', async () => {
+      await whitelist
+        .connect(owner)
+        .addRessourceToWhitelist(await addr1.getAddress());
+      await expect(
+        whitelist
+          .connect(addr1)
+          .remRessourceToWhitelist(await addr1.getAddress())
+      ).to.be.revertedWith('Ownable: caller is not the owner');
+    });
+  });
+
+  describe('Events', function () {
+    it('should emit KeyPurposeUpdate event when a dapp address is added', async () => {
+      const paddedAddress =
+        '0x' + '00'.repeat(12) + (await addr1.getAddress()).slice(2);
+
+      await expect(
+        whitelist
+          .connect(owner)
+          .addRessourceToWhitelist(await addr1.getAddress())
+      )
+        .to.emit(whitelist, 'KeyPurposeUpdate')
+        .withArgs(ethers.hexlify(paddedAddress), 4, true);
+    });
+
+    it('should emit KeyPurposeUpdate event when a dapp address is removed', async () => {
+      const paddedAddress =
+        '0x' + '00'.repeat(12) + (await addr1.getAddress()).slice(2);
+
+      await whitelist
+        .connect(owner)
+        .addRessourceToWhitelist(await addr1.getAddress());
+      await expect(
+        whitelist
+          .connect(owner)
+          .remRessourceToWhitelist(await addr1.getAddress())
+      )
+        .to.emit(whitelist, 'KeyPurposeUpdate')
+        .withArgs(ethers.hexlify(paddedAddress), 4, false);
+    });
+  });
+});
```

Comment:
```suggestion
/* tslint:disable */
import { expect } from 'chai';
import { ethers } from 'hardhat';
import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';

describe('IExecWhitelist', function () {
  async function deploySCFixture() {
    const [owner, addr1, addr2] = await ethers.getSigners();
    // Create an instance of DatasetRegistry using the address
    const IExecWhitelistFactory = await ethers.getContractFactory(
      'IExecWhitelist'
    );
    const iExecWhitelist = await IExecWhitelistFactory.deploy();
    await iExecWhitelist.deployed();

    return { iExecWhitelist, owner, addr1, addr2 };
  }

  describe('Whitelisting', () => {
    it('should allow the owner to add a dapp address to the iExecWhitelist', async () => {
      const { iExecWhitelist, owner, addr1 } = await loadFixture(deploySCFixture);
      await iExecWhitelist
        .connect(owner)
        .addRessourceToWhitelist(await addr1.getAddress());
      const paddedAddress =
        '0x' + '00'.repeat(12) + (await addr1.getAddress()).slice(2);
      expect(
        await iExecWhitelist.keyHasPurpose(ethers.utils.hexlify(paddedAddress), 4)
      ).to.equal(true);
    });

    it('should prevent non-owners from adding a dapp address to the iExecWhitelist', async () => {
      const { iExecWhitelist, addr2, addr1 } = await loadFixture(deploySCFixture);
      await expect(
        iExecWhitelist
          .connect(addr1)
          .addRessourceToWhitelist(await addr2.getAddress())
      ).to.be.revertedWith('Ownable: caller is not the owner');
    });

    it('should allow the owner to remove a dapp address from the iExecWhitelist', async () => {
      const { iExecWhitelist, owner, addr1 } = await loadFixture(deploySCFixture);
      await iExecWhitelist
        .connect(owner)
        .addRessourceToWhitelist(await addr1.getAddress());
      await iExecWhitelist
        .connect(owner)
        .remRessourceToWhitelist(await addr1.getAddress());

      const paddedAddress =
        '0x' + '00'.repeat(12) + (await addr1.getAddress()).slice(2);
      expect(
        await iExecWhitelist.keyHasPurpose(ethers.utils.hexlify(paddedAddress), 4)
      ).to.equal(false);
    });

    it('should prevent non-owners from removing a dapp address from the iExecWhitelist', async () => {
      const { iExecWhitelist, owner, addr1 } = await loadFixture(deploySCFixture);
      await iExecWhitelist
        .connect(owner)
        .addRessourceToWhitelist(await addr1.getAddress());
      await expect(
        iExecWhitelist
          .connect(addr1)
          .remRessourceToWhitelist(await addr1.getAddress())
      ).to.be.revertedWith('Ownable: caller is not the owner');
    });
  });

  describe('Events', function () {
    it('should emit KeyPurposeUpdate event when a dapp address is added', async () => {
      const { iExecWhitelist, owner, addr1 } = await loadFixture(deploySCFixture);
      const paddedAddress =
        '0x' + '00'.repeat(12) + (await addr1.getAddress()).slice(2);

      await expect(
        iExecWhitelist
          .connect(owner)
          .addRessourceToWhitelist(await addr1.getAddress())
      )
        .to.emit(iExecWhitelist, 'KeyPurposeUpdate')
        .withArgs(ethers.utils.hexlify(paddedAddress), 4, true);
    });

    it('should emit KeyPurposeUpdate event when a dapp address is removed', async () => {
      const { iExecWhitelist, owner, addr1 } = await loadFixture(deploySCFixture);
      const paddedAddress =
        '0x' + '00'.repeat(12) + (await addr1.getAddress()).slice(2);

      await iExecWhitelist
        .connect(owner)
        .addRessourceToWhitelist(await addr1.getAddress());
      await expect(
        iExecWhitelist
          .connect(owner)
          .remRessourceToWhitelist(await addr1.getAddress())
      )
        .to.emit(iExecWhitelist, 'KeyPurposeUpdate')
        .withArgs(ethers.utils.hexlify(paddedAddress), 4, false);
    });
  });
});
```
two changements : 
- use loadFixture which will reste the test environment each time
- use ethers from hardhat as it already export ethers